src/numeric-value-typing.js

```diff
@@ -0,0 +1,282 @@
+const failure = null;
+const baseTypes = ["percent", "length", "angle", "time", "frequency", "resolution", "flex"];
+
+const compatibleUnitGroups = [{
+  // Absolute lengths
+  units: ["cm", "mm", "Q", "in", "pt", "pc", "px"], canonicalUnit: "px", ratios: {
+    "cm": 96 / 2.54, "mm": (96 / 2.54) / 10, "Q": (96 / 2.54) / 40, "in": 96, "pc": 96 / 6, "pt": 96 / 72, "px": 1
+  }
+}, {
+  // Angle
+  units: ["deg", "grad", "rad", "turn"], canonicalUnit: "deg", ratios: {
+    "deg": 1, "grad": 360 / 400, "rad": 180 / Math.PI, "turn": 360
+  }
+}, {
+  // Time
+  units: ["s", "ms"], canonicalUnit: "s", ratios: {
+    "s": 1, "ms": 1 / 1000
+  }
+}, {
+  // Frequency
+  units: ["hz", "khz"], canonicalUnit: "hz", ratios: {
+    "hz": 1, "khz": 1000
+  }
+}, {
+  // Resolution
+  units: ["dpi", "dpcm", "dppx"], canonicalUnit: "dppx", ratios: {
+    "dpi": 1 / 96, "dpcm": 2.54 / 96, "dppx": 1
+  }
+}];
+
+const unitToCompatibleUnitsMap = new Map();
+for (const group of compatibleUnitGroups) {
+  for (const unit of group.units) {
+    unitToCompatibleUnitsMap.set(unit, group);
+  }
+}
+
+export function getSetOfCompatibleUnits(unit) {
+  return unitToCompatibleUnitsMap.get(unit);
+}
+
+function productOfUnionMaps(units1, units2) {
```

Comment:
Comments within the method are helpful.  Can you also provide a comment above the function declaration providing a high level description of the purpose of the function perhaps with an example of sample input and expected output.

Reply:
Comment applies to the following methods as well. Though most of the others seem to be pretty clear, a function level description (perhaps with a spec link) would nonetheless be helpful.

Reply:
Added a function level description with a link to the spec for each.